test/validation/tests/validation.go

```diff
@@ -66,5 +67,73 @@ var _ = Describe("validation", func() {
 		})
+
+		It("should be able to set nodeSelector", func() {
+			ptpConfig, err := testclient.Client.PtpV1Interface.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Get(context.Background(), testutils.PtpConfigOperatorName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Set nodeSelector to {}")
+			nodeEmpty := make(map[string]string)
+			ptpConfig.Spec.DaemonNodeSelector = nodeEmpty
+			_, err = testclient.Client.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Update(context.Background(), ptpConfig, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ptpConfig.Spec.DaemonNodeSelector).Should(Equal(nodeEmpty), "empty")
+
+			By("Set nodeSelector to {foo: bar}")
+			nodeSelected := map[string]string{"foo": "bar"}
+			ptpConfig.Spec.DaemonNodeSelector = nodeSelected
+			_, err = testclient.Client.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Update(context.Background(), ptpConfig, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			ptpConfig, err = testclient.Client.PtpV1Interface.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Get(context.Background(), testutils.PtpConfigOperatorName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ptpConfig.Spec.DaemonNodeSelector).Should(Equal(nodeSelected), "foobar")
+
+			By("Set nodeSelector back to {}")
+			ptpConfig, err = testclient.Client.PtpV1Interface.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Get(context.Background(), testutils.PtpConfigOperatorName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			ptpConfig.Spec.DaemonNodeSelector = nodeEmpty
+			_, err = testclient.Client.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Update(context.Background(), ptpConfig, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ptpConfig.Spec.DaemonNodeSelector).Should(Equal(nodeEmpty), "empty")
+		})
+
+		It("should reject event config", func() {
+			By("If TransportHost and storageType are not set")
+			ptpConfig, err := testclient.Client.PtpV1Interface.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Get(context.Background(), testutils.PtpConfigOperatorName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			ptpConfig.Spec.EventConfig = &ptpv1.PtpEventConfig{
+				EnableEventPublisher: true,
+			}
+			_, err = testclient.Client.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Update(context.Background(), ptpConfig, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+			Expect(err).Should(HaveOccurred())
+			Expect(err).To(MatchError(ContainSubstring("storageType must be set")))
+		})
+
+		It("should reject event config", func() {
+			By("If TransportHost not set and StorageType are invalid")
+			invalidStorageClass := "_invalid_storage_class"
+			ptpConfig, err := testclient.Client.PtpV1Interface.PtpOperatorConfigs(testutils.PtpLinuxDaemonNamespace).Get(context.Background(), testutils.PtpConfigOperatorName, metav1.GetOptions{})
```

Comment:
the test fails sometimes on my setup because the get still returns an old version of the operatorptpconfig. I suggest adding a small wait here (and other 2 test) to let the k8s refresh:
time.Sleep(2*time.Second)

Reply:
done